truss/templates/base.Dockerfile.jinja

```diff
@@ -9,10 +9,18 @@ FROM {{ base_image_name_and_tag }} AS truss_server
 ENV PYTHON_EXECUTABLE="{{ python_executable }}"
 
 {%- set UV_VERSION = "0.7.19" %}
-{# Use the python executable currently on the path, will respect activated virtual envs. #}
-{# We use `unsafe-best-match` since `uv` is stricter about having multiple registries, but sometimes #}
-{# we need to search for versions across multiple. #}
-{%- set sys_pip_install_command = "uv pip install --index-strategy unsafe-best-match --python " + python_exec_path %}
+{#
+NB(nikhil): We use a semi-complex uv installation command across the board:
+- A generous UV_HTTP_TIMEOUT for packages that take a long time to install.
+- We use `unsafe-best-match` since `uv` is stricter about having multiple registries, and we often need to search multiple.
+- Use the python executable currently on the path, will respect activated virtual envs.
+#}
+{%- set sys_pip_install_command =
+    "UV_HTTP_TIMEOUT=${UV_HTTP_TIMEOUT:-300} " +
```

Comment:
Intentionally chose not to set an `ENV` variable in our Dockerfile since that could theoretically alter the actually running environment for the container. I do want to derive from a user override if applicable though.